spinnaker_graph_front_end/examples/hello_world_untimed/src/hello_world.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2017-2019 The University of Manchester
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+//! imports
+#include "spin1_api.h"
+#include "common-typedefs.h"
+#include <data_specification.h>
+#include <recording.h>
+#include <simulation.h>
+#include <debug.h>
+
+//! Where we are and where to go in the simulation
+static uint32_t step = 0;
+static uint32_t n_steps = 0;
+static uint32_t run_forever = 0;
+
+//! The recording flags
+static uint32_t recording_flags = 0;
+
+//! The data
+typedef struct char_data {
+    uint32_t max_index;
+    uint32_t n_chars;
+    uint8_t chars[];
+} char_data;
+static char_data *c_data;
+
+//! human readable definitions of each region in SDRAM
+typedef enum regions_e {
+    SYSTEM_REGION,
+    PARAMS_REGION,
+    RECORDED_DATA
+} regions_e;
+
+
+//! values for the priority for each callback
+typedef enum callback_priorities {
+    SDP = 0,
+    DMA = 1
+} callback_priorities;
+
+// -------------------------------------------------------------------
+
+static void record_data(void) {
+    bool recorded = recording_record(0, c_data->chars, c_data->n_chars * sizeof(char));
+    if (!recorded) {
+        log_error("Data was not recorded...");
+    }
+}
+
+
+//! \brief Initialises the recording parts of the model
+//! \return True if recording initialisation is successful, false otherwise
+static bool initialise_recording(void) {
+    data_specification_metadata_t *data = data_specification_get_data_address();
+    void *recording_region = data_specification_get_region(RECORDED_DATA, data);
+    bool success = recording_initialize(&recording_region, &recording_flags);
+    return success;
+}
+
+static void resume_callback(void);
+
+static void run(uint unused0, uint unused1) {
+    use(unused0);
+    use(unused1);
+
+    // Increment the step as it will start at "-1"
+    step++;
+
+    log_info("Running from step %u until %u or %u", step, n_steps, c_data->max_index);
+    while ((run_forever || step < n_steps) && step < c_data->max_index) {
```

Comment:
so after the repeats be done. it does nowt. def over allcoating sdram in the python.

Reply:
It did by design, but I have removed this to avoid confusion.  No over allocation happened as explained elsewhere.